src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace NServiceBus
+{
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extentions for <see cref="EndpointConfiguration"/>.
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="endpointName">The endpoint name to be used.</param>
+        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
+        {
+            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
```

Comment:
> Does this indicate we should actually use a constant defined by core?

I think we tried to keep the constants internal whenever possible and using a constant wouldn't save us from potential breaking changes if we change the constants value? Since this is a hack into the core, I think we should just make sure we test it. I think the cleaner approach would be a custom extensibility API for the host which actually changes the parameter on the EndpointConfiguration ctor parameter, but that obviously couldn't be an extension on EndpointConfiguration.